src/resolvable.ts

```diff
@@ -0,0 +1,74 @@
+import type { DIDResolutionResult, Resolver } from 'did-resolver'
+import { createJWE, verifyJWS, resolveX25519Encrypters } from 'did-jwt'
+import type { JWE } from 'did-jwt'
+import { prepareCleartext } from 'dag-jose-utils'
+
+import { fromDagJWS, base64urlToJSON } from './utils'
+import type { DagJWS } from './utils'
+
+export interface Resolvable {
```

Comment:
Instead of a new `Resolvable` interface, could we-use the types from [did-resolver](https://github.com/decentralized-identity/did-resolver/blob/master/src/resolver.ts)?  Maybe the `DID` class could extend `Resolver`?  That way you can ask for a `Resolver` when that's the only functionality you need, but could provide a `DID` instance directly without having to reach into it for its underlying `Resolver`?

Reply:
Yeah I didn't know did-resolver already had an interface for it, so I updated the DID class to implement this interface 👍